src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,75 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import { Filter } from '../../enums/Filter';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  filterValue: string,
+  query: string,
+  onSetQuery: (value: string) => void,
+  onSelectFilter: (value: Filter) => void,
+  onApplyQuery: (value: string) => void,
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+export const TodoFilter: React.FC<Props> = ({
+  filterValue,
+  query,
+  onSetQuery,
+  onSelectFilter,
+  onApplyQuery,
+}) => {
+  const handleSelectChange = (e: React.BaseSyntheticEvent) => {
+    onSelectFilter(e.target.value);
+  };
+
+  const handleInputChange = (e: React.BaseSyntheticEvent) => {
+    onSetQuery(e.target.value);
+    onApplyQuery(e.target.value);
```

Comment:
do you really need to use these 2 functions here? it would be better to use `debounce` in `onSetQuery` only. please change the logic

Reply:
I did it by example from the video. Misha explains it in this video - https://mate.academy/learn/react/advanced-hooks?section=video&videoId=770